bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/VehicleHandler.java

```diff
@@ -0,0 +1,576 @@
+import org.openhab.binding.mercedesme.internal.utils.Mapper;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.RawType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.storage.Storage;
+import org.openhab.core.storage.StorageService;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.CommandOption;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VehicleHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class VehicleHandler extends BaseThingHandler {
+    private static final DateTimeFormatter DATE_INPUT_PATTERN = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss");
+    private static final String EXT_IMG_RES = "ExtImageResources_";
+    private static final String INITIALIZE_COMMAND = "Initialze";
+
+    private final Logger logger = LoggerFactory.getLogger(VehicleHandler.class);
+    private final Map<String, Long> timeHash = new HashMap<String, Long>();
+    private final MercedesMeCommandOptionProvider mmcop;
+    private final MercedesMeStateOptionProvider mmsop;
+    private final StorageService storageService;
+    private final HttpClient httpClient;
+    private final String uid;
+
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+    private Optional<AccountHandler> accountHandler = Optional.empty();
+    private Optional<QuantityType<?>> rangeElectric = Optional.empty();
+    private Optional<Storage<String>> imageStorage = Optional.empty();
+    private Optional<VehicleConfiguration> config = Optional.empty();
+    private Optional<QuantityType<?>> rangeFuel = Optional.empty();
+    private LocalDateTime nextRefresh;
+    private boolean online = false;
+
+    public VehicleHandler(Thing thing, HttpClient hc, String uid, StorageService storageService,
+            MercedesMeCommandOptionProvider mmcop, MercedesMeStateOptionProvider mmsop) {
+        super(thing);
+        httpClient = hc;
+        this.uid = uid;
+        this.mmcop = mmcop;
+        this.mmsop = mmsop;
+        this.storageService = storageService;
+        // https://github.com/jetty-project/jetty-reactive-httpclient/issues/33
+        httpClient.getProtocolHandlers().remove(WWWAuthenticationProtocolHandler.NAME);
```

Comment:
For clarity you could consider doing this in the handler factory, since this is where the `HttpClient` object is created, and you are sharing it with `AccountHandler` also.

Reply:
Right, shifted to Handlerfactory

Reply:
Now I also remembered this, which is related to the same problem:
https://github.com/openhab/openhab-addons/blob/0a79b8574c8441877baa13ed40240634e366c633/bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/IndegoController.java#L271-L286

I ended up with this solution recently, so sharing it as an alternative, or to get feedback if you see any problems with this. Now when considering both, your solution seems more elegant, but on the other hand my "work-around" approach makes it possible to use the shared http client.